### PR TITLE
Extensions, Microfrontends, and Menus API

### PR DESCRIPTION
**Pull request by JoelForamitti**
This PR adds first functionalities for extensions. 
- A UX extension that provides an API call to load menu items.
- A way for extensions to include themselves in the URL patterns.
- A way to build microfrontend components and serve them as static files.

### CODE REVIEW

**Comment by JoelForamitti** [comment]
There is now a webcomponent in the test_extension. If yarn build is used, the dist files are copied to the static folder. In debug mode, the static files can be accessed e.g. via `localhost:8000/static/test_extension/ux/dist/index.html`.

**Comment by JoelForamitti** [comment]
There is now a new menu API to load menu items at `/api/collectivo.ux/v1/menu/{menu_name}`.
There is currently one menu `main_menu` which we can use.

**Comment by JoelForamitti** [comment]
There is now an API to manage extensions and microfrontends

**Comment by JoelForamitti** [comment]
There is now an API to manage menus

File: collectivo-test-app/test_extension/menus.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"""Custom menu item of the test_extension."""
+# from collectivo.ux.menus import main_menu, MenuItem
```
Comment (nanu-c) [inline collectivo-test-app/test_extension/menus.py:2]: Isn't it working or why is the menu commented?

File: collectivo/urls.py [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
          SpectacularSwaggerView.as_view(url_name='collectivo:api-schema'),
          name='api-docs'),
 ]
+
+for app in settings.INSTALLED_APPS:
+    if app.startswith('collectivo.'):
+        pattern = path('', include(f'{app}.urls'))
+        urlpatterns.append(pattern)
+
+if settings.DEBUG:
```
Comment (nanu-c) [inline collectivo/urls.py:29]: static files only when `DEBUG` is true? Shouldn't we rename it to something else?

**Review by nanu-c — CHANGES_REQUESTED** [review]
Well done, just 2 little remarks

File: collectivo/urls.py (hunk 2 above)
Comment (JoelForamitti) [reply collectivo/urls.py:29]: they say in the django docs that this trick should only be used for development, and that the static files should be served over e.g. the nginx server in production

yes, we can rename it if you want!

Comment (nanu-c) [reply collectivo/urls.py:29]: Ah , yes, I have already done this here https://github.com/MILA-Wien/collectivo/blob/main/docker-compose.yml#L38

Comment (nanu-c) [reply collectivo/urls.py:29]: But i would rename it to development or smth. like this.

File: collectivo-test-app/test_extension/menus.py (hunk 1 above)
Comment (JoelForamitti) [reply collectivo-test-app/test_extension/menus.py:2]: ah, this file is deprecated. the menu is now added via API call in apps.py. I will clean up the test_extension in a future PR.

**Review by nanu-c — APPROVED** [review]

**Pull request merged**